.github/workflows/conda.yml

```diff
@@ -0,0 +1,59 @@
+name: Build conda nightlies
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+# When this workflow is queued, automatically cancel any previous running
+# or pending jobs from the same branch
+concurrency:
+  group: conda-${{ github.head_ref }}
+  cancel-in-progress: true
+
+# Required shell entrypoint to have properly activated conda environments
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  conda:
+    name: Build (and optionally upload) conda nightlies
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          use-mamba: true
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          mamba install boa conda-verify
+
+          which python
+          pip list
+          mamba list
+      - name: Build conda package
+        run: |
+          # suffix for nightly package versions
+          export VERSION_SUFFIX=a`date +%y%m%d`
+
+          conda mambabuild continuous_integration/recipe \
+                           --no-anaconda-upload \
+                           --output-folder .
+      - name: Upload conda package
+        if: |
+          github.event_name == 'push'
+          && github.ref == 'refs/heads/main'
+          && github.repository == 'dask/dask'
```

Comment:
This is uploading with every commit to `main`, but we're calling it a "nightly" which, to me, sounds more like a single nightly upload on a cron job. This could be too nitpicky of a point, but we might want to avoid the term "nightly" here. cc @jakirkham for thoughts on if you think this will lead to any confusion or not. Feel free to ignore this comment if you think the current setup is fine as is.

Reply:
Happy to use a different term here - would something like alpha or pre-release be less confusing?

Reply:
Yeah I don't have strong feelings on the name. We called them nightlies in Dask-SQL, but as you say that is not really accurate. Maybe snapshots?

Reply:
Eh, I'll retract my comment. Let's just stick with nightly for now. We can always update later if needed

Reply:
Yeah naming is hairy business 😅 Agree we can always follow up later 🙂